src/modules/c.rs

```diff
@@ -0,0 +1,220 @@
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::c::CConfig;
+use crate::formatter::StringFormatter;
+use crate::formatter::VersionFormatter;
+
+use once_cell::sync::Lazy;
+use std::borrow::Cow;
+use std::ops::Deref;
+
+/// Creates a module with the current c version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("c");
+    let config: CConfig = CConfig::try_load(module.config);
+    let is_c_project = context
+        .try_begin_scan()?
+        .set_extensions(&config.detect_extensions)
+        .set_files(&config.detect_files)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_c_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        let c_compiler_info = Lazy::new(|| {
+            context.exec_cmds_return_first(&[
+                // the compiler is usually cc, and --version works on gcc and clang
+                &["cc", "--version"],
+                // but on some platforms gcc is installed as *gcc*, not cc
+                &["gcc", "--version"],
+                // for completeness, although I've never seen a clang that wasn't cc
+                &["clang", "--version"],
+            ])
+        });
+
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "compiler_name" => {
+                    let c_compiler_info = &c_compiler_info.deref().as_ref()?.stdout;
+
+                    let c_compiler = if c_compiler_info.contains("clang") {
+                        "clang"
+                    } else if c_compiler_info.contains("Free Software Foundation") {
+                        "gcc"
+                    } else {
+                        "Unknown compiler"
+                    };
+                    Some(c_compiler).map(Cow::Borrowed).map(Ok)
+                }
+                "compiler_version" => {
+                    let c_compiler_info = &c_compiler_info.deref().as_ref()?.stdout;
+                    if config.format.contains("$compiler_version")
+                        && (c_compiler_info.contains("clang")
+                            || c_compiler_info.contains("Free Software Foundation"))
```

Comment:
If basically all -nix C compilers follow this convention, it might be better not to gate it like this.

Reply:
I wouldn't want to make that assumption, I think it's more a happy accident that clang and gcc match.

Reply:
This prompted me to look at a really old version of gcc. It doesn't fit the pattern. Bother, I'm going to need to fix that:

```
$ cc --version
cc (GCC) 3.3.5 (Debian 1:3.3.5-13)
Copyright (C) 2003 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Reply:
Fixed by now looking for the first whitespace-seperated word that looks like valid semver. Which *should* also work for a lot more compilers, so now I've also removed that condition.